packages/fiori/src/BarcodeScannerDialog.ts

```diff
@@ -120,6 +120,31 @@ type BarcodeScannerDialogScanErrorEventDetail = {
 })
 
 class BarcodeScannerDialog extends UI5Element {
+	/**
+	 * Indicates whether the dialog is open.
+	 *
+	 * @public
+	 * @default false
+	 * @since 1.23.0
+	 */
+	get open(): boolean {
+		return this._open;
+	}
+
+	set open(value: boolean) {
+		const oldValue = this._open;
+
+		if (oldValue !== value) {
+			this._open = value;
+
+			if (value) {
+				this.show();
```

Comment:
Could you try to deprecate show and close in favour of `open` property, and make `open` a regular metadata property.

See `ColorPalletePopover` for reference - 
previously it used to have a `showAt` method (it's kept for compatibility), then we introduced `open` and `opener` and deprecated `showAt`.

Reply:
Thank you for your comment. I knew that the way I've done it in this pull request is not one of the best practices, so I appreciate your guidance on this.